src/main/java/javaslang/collection/Iterator.java

```diff
@@ -429,4 +505,36 @@ public T next() {
         }
     }
 
+    class ConcatIterator<T> implements Iterator<T> {
+
+        private final Iterator<? extends Iterator<? extends T>> iterators;
+        private Iterator<? extends T> current;
+
+        private ConcatIterator(Iterator<? extends Iterator<? extends T>> iterators) {
+            this.current = Iterator.empty();
+            this.iterators = iterators;
+        }
+
+        private static boolean maybeEmpty(Iterable<?> iterable) {
+            return (iterable instanceof TraversableOnce && ((TraversableOnce<?>) iterable).isEmpty())
```

Comment:
Yes, let's delete this method and substitue the call with Sebastian's suggestion